### PR TITLE
Adds `Labels` submodule to `BatVect`.

### PR DESCRIPTION
**Pull request by theindigamer**
* Partial fix for issue #760.
* Also fixes a few typos in the `BatVect` documentation.

### CODE REVIEW

**Comment by theindigamer** [comment]
Before my commit, `make coverage` was already failing with 

```
File "qtest/all_tests.ml", line 3, characters 5-10:
Error: Unbound module OUnit
``` 
although `make test` worked just fine after downgrading to `qcheck.0.5.3.1`.

**Comment by UnixJunkie** [comment]
Hello, thanks for the pull request.
Did you follow another module to choose the label names?
What is the meaning of ~c?
x0 is called init in BatList.Labels, for example.

**Comment by theindigamer** [comment]
So I used `x0` for the folding accumulator as it was written [here](https://github.com/ocaml-batteries-team/batteries-included/wiki/Developers-guidelines) that you use `x0`. I had initially written it as `init` like BatList.Labels but changed it after reading the guidelines.

The `c` means character/cell, I suppose, as most of the existing documentation for BatVect uses `c` wherever a value of type `'a` is needed. For example, see the functions `make`, `append`, `prepend` and `set`. I also changed it to `c` in `destructive_set` for consistency. I suppose you could also use `e` for element everywhere instead of `c`.

Of course, this isn't a big deal so let me know what names are appropriate.

Also, is `make coverage` not to be used?

**Comment by UnixJunkie** [comment]
OK, I am just annoyed by the x0 now: is any Label module using it? I see ~init in BatList and BatArray.
x0 doesn't say anything to me. Ok it is short but not very meaningful.

I never used make coverage. If it is broken, you could open another issue. make test was repaired very recently in another pull request that was accepted already.

**Comment by gasche** [comment]
I agree that `x0` is not very nice. I would guess that the guidelines were written long ago and that we have not revisited this aspect. Using `~init` for consistency with List/Array seems reasonable to me, and you could update the guidelines in passing.

**Comment by theindigamer** [comment]
Alright, I'll make those changes. Should I keep the `c` as is? Or use something else?

File: src/batVect.ml [hunk 1]
```diff
@@ -1165,4 +1197,36 @@ struct
   let print ?(first="[|") ?(last="|]") ?(sep="; ") print_a out t =
     BatEnum.print ~first ~last ~sep print_a out (enum t)
 
+  module Labels =
+  struct
+    let init n ~f      = init n f
+    let concat ~v1 ~v2 = concat v1 v2
```
Comment (gasche) [inline src/batVect.ml]: I believe that the labels are not useful nor very informative for `concat`; the spatial position of the arguments should tell us everything we need to know.

**Comment by UnixJunkie** [comment]
I personally prefer 'e', standing for 'element' over 'c'.

File: src/batVect.ml (hunk 1 above)
Comment (theindigamer) [inline src/batVect.ml]: That's true but sometimes you might want to flip the arguments around. Say you created a vector using a chain of pipe operations and want to prepend it to another one. The labels are optional. I'll remove it if you don't want it.

Comment (theindigamer) [inline src/batVect.ml]: Regarding leaving out functions, I've mostly included all but some like `insert` and `remove` were not included as the data structure is the last argument anyways and there doesn't seem much reason to use the labelled version (at least to me). Again, I'll put it in if you say so.

**Comment by thizanne** [comment]
I believe the point of adding labels is to improve readability in exchange for a little bit of verbosity.  Thus why not use `~elem` rather than "`e` that stands for element" ? `f` is a common one-letter name for a function, `e` is not as much widespread. I'm all for being more explicit.

**Comment by gasche** [comment]
I am ok with the current state. @UnixJunkie, @thizanne, is there a consensus to merge?

**Comment by UnixJunkie** [comment]
I'm OK. Thanks to @theindigamer for having improved his patch.

**Comment by theindigamer** [comment]
Cheers :smile:

**Comment by thizanne** [comment]
Merged, thanks !

**Comment by thizanne** [comment]
Sorry, I just read everyone was OK and went a bit fast. We should have a Changes entry and probably some tests -- I will do the work.

**Pull request merged**